api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -20,27 +20,26 @@
 
 /**
  * An abstract event envelope
- * 
- * @author fabiojose
- *
  * @param <A> The attributes type
  * @param <T> The 'data' type
+ * @author fabiojose
  */
 public interface CloudEvent<A extends Attributes, T> {
 
-	/**
-	 * The event context attributes
-	 */
-	A getAttributes();
-	
-	/**
-	 * The event data
-	 */
-	Optional<T> getData();
-	
-	/**
-	 * The event extensions
-	 */
-	Map<String, Object> getExtensions();
-	
+    /**
+     * The event context attributes
+     */
+    A getAttributes();
+
+    /**
+     * The event data
+     */
+    Optional<T> getData();
+
+    byte[] getDataBase64();
```

Comment:
Why we can not use `Optional<T> getData()`?

Why an attribute called getDataBase64 exclusively to work with base64?

Reply:
Because there's no way to assign a `byte[]` to `data` as it is using generics. In the tests you can see that instead of declaring `Optional<byte[]>` I declare `Optional<String>`. Theoretically the `byte[]` should be a binary representation of the String and maybe the SDK consumer might want to create the `String` from the `byte[]`.

When receiving a structured CloudEvent with content type being `text/plain`, it can include a `data_base64` that I won't be able to easily unmarshal into a `CloudEvent<? extends Attributes, String>` that should also work for a CloudEvent with `data` being just text. Do you see my point? Am I missing something?

Thanks for your comments.